plugins/filter/filters.py

```diff
@@ -32,24 +32,37 @@ def filters(self):
             'resolve_principal': self.resolve_principal
         }
 
-    def normalize_sasl_protocol(self, protocol):
-        # Returns standardized value for sasl mechanism string
-        normalized = 'GSSAPI' if protocol.lower() == 'kerberos' \
-            else 'SCRAM-SHA-512' if protocol.upper() == 'SCRAM' \
-            else 'SCRAM-SHA-256' if protocol.lower() == 'scram256' \
-            else 'PLAIN' if protocol.upper() == 'PLAIN' \
-            else 'OAUTHBEARER' if protocol.upper() == 'OAUTH' \
-            else 'none'
-        return normalized
+    def normalize_sasl_protocol(self, protocols):
```

Comment:
This can be either refactored to be renamed as normalize_sasl_protocols or maybe we can keep this function unchanged and from wherever we were calling it from there we can add a loop to call it multiple times